src/Microsoft.AspNetCore.Authentication/RemoteAuthenticationHandler.cs

```diff
@@ -87,6 +87,12 @@ public abstract class RemoteAuthenticationHandler<TOptions> : AuthenticationHand
             return true;
         }
 
+        /// <summary>
+        /// Authenticate the user identity with the identity provider. 
+        ///
+        /// This could be done through a back channel communication with the identity provider. Exception thrown during
+        /// the authenticating should be saved to the AuthenticateResult.
```

Comment:
The .NET Exception guidelines are a useful reference: https://msdn.microsoft.com/en-us/library/ms229005(v=vs.100).aspx
The relevant bit here is to avoid catching too broad a range of exception types. E.g. only catch exceptions when you know what causes them.